src/backy/utils.py

```diff
@@ -136,27 +137,25 @@ def __exit__(self, exc_type=None, exc_value=None, exc_tb=None):
 
     def open_new(self, mode):
         """Open this as a new (temporary) file that will be renamed on close."""
-        assert not self.f
+        assert self.f is None
         self.f = tempfile.NamedTemporaryFile(
             mode, dir=os.path.dirname(self.filename), delete=False
         )
         self.rename = True
 
     def open_copy(self, mode):
         """Open an existing file, make a copy first, rename on close."""
-        assert not self.f
-
+        self.open_new("wb")
```

Comment:
hmhmn... why? is this do avoid race conditions? this sounds like there might be a crash-based unclean path somewhere ...

Reply:
So, as discussed in person: there was a bug in the previous version if the original file did not yet exist.